### PR TITLE
Mat 1138 add to stake

### PR DESCRIPTION
**Pull request by imyourm8**

### CODE REVIEW

**Comment by imyourm8** [comment]
I will merge this branch after https://github.com/maticnetwork/contracts/pull/250 is merged.

File: contracts/staking/stakeManager/StakeManager.sol [hunk 1]
```diff
@@ -165,14 +170,22 @@ contract StakeManager is IStakeManager, StakeManagerStorage {
...
         perceivedStake = Math.max(
             perceivedStake,
             validatorAuction[validatorId].amount
         );
+        
+        uint256 finalBid = bid;
+        
+        if (senderValidatorId != 0) {
+            // if current validator bids, increase bid by current stake
+            finalBid = perceivedStake.add(bid);
```
Comment (jdkanani) [inline contracts/staking/stakeManager/StakeManager.sol]: What will happen when someone overbids a validator's bid? Will validator be able to instantly withdraw whole stake or just extra bid amount?

Comment (imyourm8) [reply contracts/staking/stakeManager/StakeManager.sol]: Whole stake.

Comment (imyourm8) [reply contracts/staking/stakeManager/StakeManager.sol]: Realized there is not such test case, adding

**Comment by 0xAshish** [comment]
@imyourm8 I am working on this for outflow for the owner of slot.

**Pull request closed without merging**